packages/replay-next/components/sources/hooks/useSourceSearch.ts

```diff
@@ -125,23 +125,40 @@ export default function useSourceSearch(
     onChangeDispatching
   );
 
+  const { search: dispatchSearch } = dispatch;
+
+  const memoizedSearch = useCallback(
+    (query: string) => dispatchSearch(query, modifiers),
+    [dispatchSearch, modifiers]
+  );
+
+  const memoizedSetModifiers = useCallback(
+    (newModifiers: SearchModifiers) => {
+      if (isEqual(modifiers, newModifiers)) {
+        return;
+      }
+
+      setModifiers(newModifiers);
+      dispatchSearch(state.query, newModifiers);
+    },
+    [dispatchSearch, modifiers, state.query]
+  );
+
+  const memoizedSetScope = useCallback(
+    (sourceId: SourceId | null, code: string) => {
+      setScope({ code, sourceId });
+    },
+    [setScope]
+  );
```

Comment:
This is just an optimization. I noticed `SourceSearchContext` effect was firing too often– because the `setScope` function inside of `memoizedActions` was being re-created every time `state.query` changed, even though the only value it uses (`setScope`) is stable.